### ISSUE TITLE
[Bug]: Unable to export application

### ISSUE THREAD

**Issue by rajkumarks7** [appsmithorg/appsmith#31672]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

backend stdout | WARNING: An illegal reflective access operation has occurred
backend stdout | WARNING: Illegal reflective access by retrofit2.Platform (jar:file:/opt/appsmith/backend/server.jar!/BOOT-INF/lib/retrofit-2.9.0.jar!/) to constructor java.lang.invoke.MethodHandles$Lookup(java.lang.Class,int)
backend stdout | WARNING: Please consider reporting this to the maintainers of retrofit2.Platform
backend stdout | WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
backend stdout | WARNING: All illegal access operations will be denied in a future release
backend stdout |
backend stdout |
rts stdout | (node:1490) Warning: Accessing non-existent property 'MongoError' of module exports inside circular dependency
rts stdout | (Use `node --trace-warnings ...` to show where the warning was created)
rts stdout |
rts stdout | RTS version v1.8.15 running at http://localhost:8091
rts stdout |
rts stdout | Watching MongoDB
rts stdout |
rts stdout | Caught exception: MongoError: $changeStream may not be opened on the internal admin database
rts stdout | Exception origin: uncaughtException
rts stdout |
rts stdout | events.js:377
rts stdout | throw er; // Unhandled 'error' event
rts stdout | ^
rts stdout |
rts stdout | MongoError: $changeStream may not be opened on the internal admin database
rts stdout | at MessageStream.messageHandler (/opt/appsmith/rts/node_modules/mongodb/lib/cmap/connection.js:268:20)
rts stdout | at MessageStream.emit (events.js:400:28)
rts stdout | at processIncomingData (/opt/appsmith/rts/node_modules/mongodb/lib/cmap/message_stream.js:144:12)
rts stdout | at MessageStream._write (/opt/appsmith/rts/node_modules/mongodb/lib/cmap/message_stream.js:42:5)
rts stdout | at writeOrBuffer (internal/streams/writable.js:358:12)
rts stdout | at MessageStream.Writable.write (internal/streams/writable.js:303:10)
rts stdout | at TLSSocket.ondata (internal/streams/readable.js:731:22)
rts stdout | at TLSSocket.emit (events.js:400:28)
rts stdout | at addChunk (internal/streams/readable.js:293:12)
rts stdout | at readableAddChunk (internal/streams/readable.js:267:9)
rts stdout | Emitted 'error' event on ChangeStream instance at:
rts stdout | at processError (/opt/appsmith/rts/node_modules/mongodb/lib/change_stream.js:571:38)
rts stdout | at ChangeStreamCursor.<anonymous> (/opt/appsmith/rts/node_modules/mongodb/lib/change_stream.js:435:5)
rts stdout | at ChangeStreamCursor.emit (events.js:400:28)
rts stdout | at /opt/appsmith/rts/node_modules/mongodb/lib/core/cursor.js:346:16
rts stdout | at /opt/appsmith/rts/node_modules/mongodb/lib/core/cursor.js:739:9
rts stdout | at /opt/appsmith/rts/node_modules/mongodb/lib/change_stream.js:329:9
rts stdout | at done (/opt/appsmith/rts/node_modules/mongodb/lib/core/cursor.js:461:7)
rts stdout | position: 20,
rts stdout | buffer: Buffer(20) [Uint8Array] [
rts stdout | 96, 204, 180, 67, 236, 7,
rts stdout | 87, 247, 144, 211, 176, 216,
rts stdout | 13, 43, 82, 13, 209, 141,
rts stdout | 126, 91
rts stdout | ]
rts stdout | },
rts stdout | keyId: Long { _bsontype: 'Long', low_: 7, high_: 1703875013 }
rts stdout | }
rts stdout | },
rts stdout | operationTime: Timestamp { _bsontype: 'Timestamp', low_: 1, high_: 1710179303 }
rts stdout | }
rts stdout |
2024-03-11 17:48:27,340 INFO exited: rts (exit status 1; not expected)
2024-03-11 17:48:27,491 INFO spawned: 'rts' with pid 1506
backend stdout | [2024-03-11 17:48:27,485] - Enabled plugins: []
backend stdout |
2024-03-11 17:48:27,493 INFO success: rts entered RUNNING state, process has stayed up for > than 0 seconds (startsecs)

### Steps To Reproduce

use the mongodb with mongo+srv connection with kubernetes

### Public Sample App

https://as.metintapi.com/applications

### Environment

Production

### Severity

Critical (Broken Production apps)

### Version

self hosted 1.8.15

**Comment by Nikhil-Nandagopal** [comment]
@rajkumarks7 hello we need more details as to what the problem is and what steps did you use to encounter it. I also see that you are on a very old version of the product and you should update to the latest version.
https://docs.appsmith.com/getting-started/setup/instance-management/update-appsmith

**Comment by rajkumarks7** [comment]
@Nikhil-Nandagopal  i am getting same error for the v 1.9.2 as well my db connection string 
looks like this 
mongodb+srv://user:password@url/admin?w=majority&retrywrites=true&authsource=admin&minpoolsize=0

**Comment by Nikhil-Nandagopal** [comment]
@rajkumarks7 could you please explain what is the problem you are facing?
What steps did you take to encounter it?

**Comment by rajkumarks7** [comment]
i am getting error 503 when i click to export or to edit the apps which are existing. already in a database

**Comment by rajkumarks7** [comment]
{"responseMeta":{"status":500,"success":false,"error":{"code":5000,"message":"Internal server error while processing request"}}}

getting this error when trying to export the app.

**Comment by rahulbarwal** [comment]
@rajkumarks7 Can you please try updating your instance to the latest version. We have made changes to the import-export logic which we think will help in this case. https://docs.appsmith.com/getting-started/setup/instance-management/update-appsmith
cc: @riteshkew 

On a side note, you have stated severity as: Critical (Broken Production apps), Can you please share more details, if it indeed is the case?

**Comment by rajkumarks7** [comment]
I am using the 1.9.7 version of appsmith

**Comment by Nikhil-Nandagopal** [comment]
@rajkumarks7 our latest release is 1.17. Please update to that release. Also ensure that you went to the checkpoint release when you did an update. I'm unclear how this broke your apps if this is only stopping the export flow for you

**Comment by rajkumarks7** [comment]
can you please give me the docker version please @Nikhil-Nandagopal

**Comment by rajkumarks7** [comment]
getting below error. @Nikhil-Nandagopal 
userEmail=eshan@looprl.com, sessionId=7170f083-4c5e-4e9c-9577-d009d9b4aa74, thread=nioEventLoopGroup-3-4, requestId=721266f9-f56a-49aa-80ec-37e7fd07243a -
backend stdout | java.lang.IllegalArgumentException: No enum constant com.appsmith.server.acl.AclPermission.CREATE_PERMISSION_GROUPS
backend stdout |        at java.base/java.lang.Enum.valueOf(Enum.java:240)
backend stdout |        Suppressed: reactor.core.publisher.FluxOnAssembly$OnAssemblyException:
backend stdout | Error has been observed at the following site(s):

**Comment by rajkumarks7** [comment]
getting below error while i am trying to export the app. @Nikhil-Nandagopal 
[2024-03-14 03:24:34,275] userEmail=eshan@looprl.com, sessionId=6cdeb918-5f9c-4c9d-8c8f-861f315ce199, thread=lettuce-epollEventLoop-6-1, requestId=450aed41-03ab-45cc-9266-a2337e46ba91 - Going to export application with id: 65e100680efa720b2b6447cb, branch: null
backend stdout | 
backend stdout | [2024-03-14 03:24:34,404] userEmail=eshan@looprl.com, sessionId=6cdeb918-5f9c-4c9d-8c8f-861f315ce199, thread=nioEventLoopGroup-3-2, requestId=450aed41-03ab-45cc-9266-a2337e46ba91 - 
backend stdout | java.lang.IllegalStateException: Unable to invoke Cipher due to bad padding
backend stdout |        at org.springframework.security.crypto.encrypt.CipherUtils.doFinal(CipherUtils.java:141)
backend stdout |        Suppressed: reactor.core.publisher.FluxOnAssembly$OnAssemblyException: 
backend stdout | Error has been observed at the following site(s):
backend stdout |        *__checkpoint ⇢ Handler com.appsmith.server.controllers.ApplicationController#getApplicationFile(String, String) [DispatcherHandler]
backend stdout | Original Stack Trace:

**Comment by rajkumarks7** [comment]
getting the below error when i try to edit the app i am the admin for the app. @Nikhil-Nandagopal 
rts stdout | ns:/page/edit # socket t6W-jP9pQwM_nlfGAAAd left the room page:65e12f590efa720b2b644815
rts stdout | 
backend stdout | [2024-03-14 03:26:54,575] userEmail=eshan@looprl.com, sessionId=6cdeb918-5f9c-4c9d-8c8f-861f315ce199, thread=reactor-http-epoll-1, requestId=febeca17-1753-46e5-9402-dd4de0c226a2 - update layout received for page undefined
backend stdout | 
backend stdout | [2024-03-14 03:26:54,606] userEmail=eshan@looprl.com, sessionId=6cdeb918-5f9c-4c9d-8c8f-861f315ce199, thread=nioEventLoopGroup-3-4, requestId=febeca17-1753-46e5-9402-dd4de0c226a2 - 
backend stdout | com.appsmith.server.exceptions.AppsmithException: Unable to find applicationId undefined. Either the asset doesn't exist or you don't have required permissions
backend stdout |        at com.appsmith.server.services.ce.LayoutActionServiceCEImpl.updateLayout(LayoutActionServiceCEImpl.java:651)
backend stdout |        Suppressed: reactor.core.publisher.FluxOnAssembly$OnAssemblyException: 
backend stdout | Error has been observed at the following site(s):
backend stdout |        *__checkpoint ⇢ Handler com.appsmith.server.controllers.LayoutController#updateLayout(String, String, String, Layout, String) [DispatcherHandler]
backend stdout | Original Stack Trace:

**Comment by rahulbarwal** [comment]
Hey @rajkumarks7, 
We would love to get on a call to understand your issue and provide solutions. Please feel free to book a slot, here is my calendly: https://calendly.com/rahulbarwal/60-minute-meeting

**Comment by rahulbarwal** [comment]
Thanks @rajkumarks7 for taking out time to talk to us over call. 
* On call, we discovered two major problems: 
* The database (DB) was in an inconsistent state. This was due switching between EE and CE.
* Instance was not starting up due to a missing license key. 
* And finally the datasources being corrupt might have been due to change in encryption salt/secret.

I am glad that we could help in resolving your issue. If you face this issue again, feel free to open a github issue. 
Have a great day.
